qt_configure.bzl

```diff
@@ -13,37 +13,70 @@ def qt_autoconf_impl(repository_ctx):
        repository_ctx: repository context
     """
     os_name = repository_ctx.os.name.lower()
-    is_linux_machine = False
-    if os_name.find("windows") != -1:
+    if os_name.find("windows") == -1 and os_name.find("linux") == -1:
+        fail("Unsupported OS: %s" % os_name)
+    is_linux_machine = os_name.find("windows") == -1
+
+    # Location of the different Qt artifacts on Linux system installed standalone as well as Windows
+    # Qt includes
+    # <location>/include
+    #
+    # Qt libraries
+    # <location>/lib
+    #
+    # uic, rcc, moc
+    # <location>/bin
+
+    # Location of the different Qt artifacts on Linux system installed with apt
+    # Qt includes
+    # /usr/include/x86_64-linux-gnu/qt5
+    #
+    # Qt libraries
+    # /lib/x86_64-linux-gnu/
+    # /usr/lib/x86_64-linux-gnu/
+    #
+    # uic, rcc, moc
+    # /usr/bin/
+
+    # If QT_DIR is defined and it exists, we consider that it is a standalone installation
+    # Only in the case that is linux and QT_DIR is not defined it is considerd a none standalone
+    # installation
+    qt_dir_path = _get_env_var(repository_ctx, "QT_DIR", None)
+    is_standalone = qt_dir_path != None and repository_ctx.path(qt_dir_path).exists
+    if not is_linux_machine:
         # Inside this folder, in Windows you can find include, lib and bin folder
         default_qt_path = "C:\\\\Qt\\\\5.9.9\\\\msvc2017_64\\\\"
-    elif os_name.find("linux") != -1:
-        is_linux_machine = True
-        # In Linux, this is the equivalent to the include folder, the binaries are located in
-        # /usr/bin/
-        # This would be the path if it has been installed using a package manager
-        default_qt_path = "/usr/include/x86_64-linux-gnu/qt5"
-        if not repository_ctx.path(default_qt_path).exists:
-            default_qt_path = "/usr/include/qt"
+        qt_prefix = default_qt_path
+        if is_standalone:
+            qt_prefix = qt_dir_path
+        qt_includes = qt_prefix + "\\\\include"
+        qt_libs = qt_prefix + "\\\\lib"
+        qt_bins = qt_prefix + "\\\\bin"
     else:
-        fail("Unsupported OS: %s" % os_name)
-
-    if repository_ctx.path(default_qt_path).exists:
-        print("Installation available on the default path: ", default_qt_path)
+        if is_standalone:
+            qt_prefix = qt_dir_path
+            qt_includes = qt_prefix + "/include"
+            qt_libs = qt_prefix + "/lib"
+            qt_bins = qt_prefix + "/bin"
```

Comment:
hello, curious about this part, how can we pass bin path to qt_toolchain rule to get moc, rcc and uic path?